tests/rptest/test_suite_infinite_retention.yml

```diff
@@ -0,0 +1,14 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+retention:
```

Comment:
nit: many tests include some kind of "retention". Maybe this should be some `tiered_storage_scale` or something generic.